src/core/server/http/http_server.mocks.ts

```diff
@@ -19,7 +19,54 @@
 import { Request, ResponseToolkit } from 'hapi';
 import { merge } from 'lodash';
 
-import { KibanaRequest } from './router';
+import querystring from 'querystring';
+
+import { schema } from '@kbn/config-schema';
+
+import { KibanaRequest, RouteMethod } from './router';
+
+interface RequestFixtureOptions {
+  headers?: Record<string, string>;
+  params?: Record<string, unknown>;
+  body?: Record<string, unknown>;
+  query?: Record<string, unknown>;
+  path?: string;
+  method?: RouteMethod;
+}
+
+function createKibanaRequestMock({
+  path = '/path',
+  headers = { accept: 'something/html' },
+  params = {},
+  body = {},
+  query,
```

Comment:
Thoughts on defaulting `query` to an empty object? I think it could simplify a little:

```diff
- query,
+ query = {},
```

```diff
+ const qs = querystring.stringify(query);
```

```diff
- query: query || {},
+ query,
```

```diff
  url: {
    path,
-    query: query ? querystring.stringify(query) : query,
+    query: qs,
-    search: query ? `?${querystring.stringify(query)}` : query,
+    search: qs && ? `?${qs}`,
  },
```